src/main/java/com/google/jenkins/plugins/k8sengine/KubernetesVerifiers.java

```diff
@@ -0,0 +1,198 @@
+      this.log = log;
+      this.status = status;
+      this.manifestObject = object;
+    }
+
+    public String toString() {
+
+      return new StringBuilder()
+          .append(
+              (status)
+                  ? Messages.KubernetesEngineBuilder_VerifyingLogSuccess(manifestObject.describe())
+                  : Messages.KubernetesEngineBuilder_VerifyingLogFailure(manifestObject.describe()))
+          .append("\n")
+          .append(log)
+          .toString();
+    }
+  }
+
+  public interface Verifier {
+    /**
+     * @param kubectl A KubectlWrapper object for querying the object type in the cluster.
+     * @return true If the resource was verified, false otherwise.
+     */
+    VerificationResult verify(KubectlWrapper kubectl, Manifests.ManifestObject object);
+  }
+
+  /*
+   Fallback verifier shouldn't be used.
+  */
+  private static class DefaultVerifier implements Verifier {
+    public VerificationResult verify(KubectlWrapper kubectl, Manifests.ManifestObject object) {
+      LOGGER.info("Reached unimplemented default verifier.");
+
+      return new VerificationResult(
+          Messages.KubernetesEngineBuilder_VerifierNotImplementedFor(object.describe()),
+          false,
+          object);
+    }
+  }
+  /** Verify that deployment has at least the minimum number of available replicas */
+  public static class DeploymentVerifier implements Verifier {
+
+    private static final String AVAILABLE_REPLICAS = "availableReplicas";
+    private static final String MINIMUM_REPLICAS_JSONPATH = "spec.replicas";
+    private static final String STATUS_JSONPATH = "status";
+
+    public VerificationResult verify(KubectlWrapper kubectl, Manifests.ManifestObject object) {
+      LOGGER.info(String.format("Verifying deployment, %s", object.getName()));
+      StringBuilder log = new StringBuilder();
+      Object json = null;
+
+      try {
+        json = kubectl.getObject(object.getKind().toLowerCase(), object.getName());
+      } catch (Exception e) {
+        return errorResult(e, object);
+      }
+
+      Integer minReplicas = JsonPath.read(json, MINIMUM_REPLICAS_JSONPATH);
+      Map<String, Object> status = JsonPath.read(json, STATUS_JSONPATH);
+      Integer availableReplicas = (Integer) status.getOrDefault(AVAILABLE_REPLICAS, 0);
+      boolean verified =
+          minReplicas != null
+              && availableReplicas != null
+              && minReplicas.intValue() <= availableReplicas.intValue();
+
+      log.append("AvailableReplicas = ")
+          .append(availableReplicas)
+          .append(",")
+          .append(" MinimumReplicas = ")
+          .append(minReplicas)
+          .append("\n");
+
+      return new VerificationResult(log.toString(), verified, object);
+    }
+  }
+
+  private static final Verifier defaultVerifier = new DefaultVerifier();
+
+  /*
+   * register verifiers
+   */
+  static {
+    verifiers.put(kindIdentifier("deployment"), new DeploymentVerifier());
```

Comment:
Prefer keeping static initializers at top of class after variables, but before constructors.